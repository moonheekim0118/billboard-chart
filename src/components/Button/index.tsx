import React from 'react';
import styled from 'styled-components';

interface Props {
    onClick: (e: any) => void;
    title: string;
    disabled: boolean;
    type: string;
}

const Button = (props: Props) => {
    return (
        <Container
            type={props.type}
            disabled={props.disabled}
            onClick={props.onClick}>
            {props.title}
        </Container>
    );
};

const Container = styled.button`
    background-color: #ecc422;
    color: #fff;
    font-size: 1.1em;
    font-weight: bold;
    border: none;
    border-top-right-radius: 10px;
    border-bottom-right-radius: 10px;
    padding: 10px 15px;

    cursor: pointer;
    transition: background-color 0.5s ease;

    &:disabled {
        background-color: #f3d972;
        cursor: not-allowed;
    }

    &:hover {
        background-color: #d3ad12;
    }
`;

export default Button;
