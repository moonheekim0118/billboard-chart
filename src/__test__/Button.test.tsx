import 'jsdom-global/register';
import React from 'react';
import Button from '../components/Button';
import { mount } from 'enzyme';

describe('<Button/>', () => {
    let container = null;
    const mockFn = jest.fn();

    it('renders correctly', () => {
        container = mount(
            <Button
                title={'sample'}
                disabled={false}
                type={'submit'}
                onClick={mockFn}
            />
        );
    });

    it('matches snapshot', () => {
        expect(container.html()).toMatchSnapshot();
    });

    it('should call onClick if Container is clicked', () => {
        container.simulate('click');
        expect(mockFn).toHaveBeenCalled();
    });
});
