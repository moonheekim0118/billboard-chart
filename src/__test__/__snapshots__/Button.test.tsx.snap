// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> matches snapshot 1`] = `"<button type=\\"submit\\" class=\\"sc-bdfBwQ cLmSPl\\">sample</button>"`;
